dhis2/dhis2-core
# feat: log X-Request-ID when available [DHIS2-6494]

### Summary
Uses `MDC` of log4j to add information on the `X-Request-ID` HTTP header to log entries.
If the header is not an alphanumeric value between 1 and 36 characters the value shown in the logs is `(illegal)`. This allows for example UUIDs as they usually are formatted as well as DHIS2 usual 11 character UIDs.

### Manual Testing
Run an API request with `X-Request-ID` set to some text and check the logs - the logged rows should end with the text provided.
